# HG changeset patch
# User Rob Mahurin <octave@tertiarysource.net>
# Date 1261525564 21600
#      Tue Dec 22 17:46:04 2009 -0600
# Node ID 3275dda012f7d38f26ac5ef36efb4e3f6e8277bc
# Parent  49563d18ca78204b5dfb251c7c27859cad85ccc2
find-defun-files.sh: use sed and egrep as found by configure

diff --git a/src/ChangeLog b/src/ChangeLog
--- a/src/ChangeLog
+++ b/src/ChangeLog
@@ -1,8 +1,12 @@
+2009-12-22  Rob Mahurin  <octave@tertiarysource.net>
+
+	* find-defun-files.sh: Use SED and EGREP as returned by configure.
+
 2009-12-20  Shai Ayal  <shaiay@users.sourceforge.net>
 
 	* DLD-FUNCTIONS/fltk_backend.cc (plot_sindow::set_name): New method.
 	(figure_manager::set_name, figure_manage::do_set_name): New methods.
 	(fltk_backend::property_changed): Handle change of figure title string.
 
 2009-12-19  Rik <octave@nomad.inbox5.com>
 
diff --git a/src/find-defun-files.sh b/src/find-defun-files.sh
--- a/src/find-defun-files.sh
+++ b/src/find-defun-files.sh
@@ -1,11 +1,13 @@
 #! /bin/sh
 
 set -e
+SED=${SED:-sed}
+EGREP=${EGREP:-egrep}
 
 # Some stupid egreps don't like empty elements in alternation patterns,
 # so we have to repeat ourselves because some stupid egreps don't like
 # empty elements in alternation patterns.
 
 DEFUN_PATTERN="^[ \t]*DEF(CONSTFUN|CMD|UN|UN_DLD|UNX_DLD|UN_TEXT)[ \t]*\\("
 
 srcdir="$1"
@@ -13,12 +15,12 @@ shift
 
 for arg
 do
   if [ -f "$arg" ]; then
     file="$arg"
   else
     file="$srcdir/$arg"
   fi
-  if [ "`egrep -l "$DEFUN_PATTERN" $file`" ]; then
-    echo "$file" | sed 's,.*/,,; s/\.\(cc\|yy\|ll\)$/.df/';
+  if [ "`$EGREP -l "$DEFUN_PATTERN" $file`" ]; then
+    echo "$file" | $SED 's,.*/,,; s/\.\(cc\|yy\|ll\)$/.df/';
   fi
 done
